CenterofExcellenceALMAccelerator/SolutionPackage/CanvasApps/cat_poweropsdevopsedition_7f3f4_DocumentUri_msapp_src/Src/DeploymentSettingsScreen.fx.yaml

```diff
@@ -3093,6 +3093,7 @@ DeploymentSettingsScreen As screen:
                         }
                     )
                 );
+                Collect(DeploymentSettings,{Name: "SolutionUpgrade",Value: If(chkUpgradeSolution.Value,"upgrade","update")});
```

Comment:
A thought on refactoring this. If instead of setting "SolutionUpgrade" to "upgrade" or "update" you instead set "TriggerSolutionUpgrade" to true or false you could remove a lot of logic both in the app and in the pipelines where you are translating between the text "upgrade" and "update" to true or false and also not have two different variables "SolutionUpgrade" and "TriggerSolutionUpgrade" in the pipeline.

Reply:
e.g. 

```
Collect(DeploymentSettings,{Name: "TriggerSolutionUpgrade",Value: chkUpgradeSolution.Value});
```

Also, since this is a Collect() have you tested what happens when you check / uncheck multiple times? Does it add duplicates to the collection?

Reply:
Refractored the code by setting true/false instead of upgrade/update. However, chkUpgradeSolution.Value returns bool and can't set in json format so used Text(chkUpgradeSolution.Value)